contracts/protocol/clients/voucher/BosonVoucher.sol

```diff
@@ -785,15 +798,59 @@ contract BosonVoucherBase is IBosonVoucher, BeaconClientBase, OwnableUpgradeable
         uint256 _tokenId,
         uint256 _offerId
     ) internal {
-        require(_from == owner(), NO_SILENT_MINT_ALLOWED);
```

Comment:
Removing this allows anyone (who has non zero balance) to simply transfer preminted vouchers, even if they are not owner of the preminted vouchers.
So if seller has two offers, one cheap and one expensive, attacker can just buy the cheaper, and then steals the expensive one without paying anything.
It must be checked that `_from` actually owns the `tokenId`

This is the test I used to validate that attack exists:
```javascript
            it.only("Transfer preminted voucher, but from is not the voucher owner", async function () {
              // make sure "rando" has non-zero balance. In reality, that would be easy to achieve by just buying something cheap.
              await bosonVoucher.connect(assistant)[selector](assistant.address, rando.address, tokenId, ...additionalArgs);

              // next token id. Make sure that assistant is the owner
              tokenId = Number(tokenId)+1
              let tokenOwner = await bosonVoucher.ownerOf(tokenId);
              assert.equal(tokenOwner, assistant.address, "Seller is not the owner");

              // Following call should fail, since rando is not the owner of preminted voucher 
              await expect(
                bosonVoucher.connect(rando)[selector](rando.address, rando.address, tokenId, ...additionalArgs)
              ).to.be.reverted;
            });
```
Put this test in the place where "Transfer preminted voucher, but from is not the contract owner" was before. It should fail, but it doesn't.

Reply:
fixed